### PR TITLE
added RepositoryException

### LINKED ISSUE #919: Add RepositoryException which is a wrapper for a catchable Exception
Currently all exceptions are catched, that are thrown inside the application fetching process of a Repository.
This approach has the problem that errors, which are repository independent, are catched too.
To make this more clear, which exceptions can or should be thrown and processed inside a repository I suggest to introduce a new `RepositoryException` class.

For more details see #913

### PR DESCRIPTION
**Pull request by plata**
fixes #919

### CODE REVIEW

File: phoenicis-repository/src/main/java/org/phoenicis/repository/RepositoryException.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+package org.phoenicis.repository;
+
+public class RepositoryException extends RuntimeException {
+    public RepositoryException(String message) {
+        super(message);
+    }
+
+    public RepositoryException(String message, Exception e) {
```
Comment (madoar) [inline phoenicis-repository/src/main/java/org/phoenicis/repository/RepositoryException.java]: I believe `public RepositoryException(String message, Throwable throwable)` is better.

Comment (plata) [inline phoenicis-repository/src/main/java/org/phoenicis/repository/RepositoryException.java]: I just copied `CabException`. I guess @qparis had a reason to implement it like this. If you still think `Throwable` is better, I can change it of course.

**Comment by plata** [comment]
I didn't see any place where this could be used in the other repository types. If you think I should use it in a particular place, please tell me where.

**Comment by madoar** [comment]
I think we should throw it in the catch blocks of the `GitRepository`:
```java
        } catch (RepositoryNotFoundException | GitAPIException e) {
            LOGGER.error(String.format("Folder '%s' is no git-repository", this.localFolder.getAbsolutePath()), e);
        } catch (IOException e) {
            LOGGER.error("An unknown error occurred", e);
        } finally {
            // close repository to free resources
            if (gitRepository != null) {
                gitRepository.close();
            }
        }
```

For the `LocalRepository` I agree, it seems to be more useful to simply ignore single scripts, applications or categories, instead of throwing an exception, which skips the whole repository.

File: phoenicis-repository/src/main/java/org/phoenicis/repository/RepositoryException.java (hunk 1 above)
Comment (madoar) [inline phoenicis-repository/src/main/java/org/phoenicis/repository/RepositoryException.java]: Take a look at the extended class https://docs.oracle.com/javase/7/docs/api/java/lang/RuntimeException.html. The constructor you call with `super(message, e);` takes a `Throwable` as its second argument.

**Pull request merged**